diff --git a/llvm/include/llvm/Config/config.h.cmake b/llvm/include/llvm/Config/config.h.cmake
index 1a38bc15ab9..9a3417f1c8c 100644
--- a/llvm/include/llvm/Config/config.h.cmake
+++ b/llvm/include/llvm/Config/config.h.cmake
@@ -333,7 +333,9 @@
 #cmakedefine RETSIGTYPE ${RETSIGTYPE}
 
 /* Define if std::is_trivially_copyable is supported */
+#if !defined(_MSC_VER) || _MSC_VER < 1927
 #cmakedefine HAVE_STD_IS_TRIVIALLY_COPYABLE ${HAVE_STD_IS_TRIVIALLY_COPYABLE}
+#endif
 
 /* Define to a function implementing stricmp */
 #cmakedefine stricmp ${stricmp}
